src/charm.py

```diff
@@ -845,12 +903,31 @@ def new_session_key():
     return {SESSION_KEY_LOOKUP: b64encode(os.urandom(64)).decode("utf-8")}
 
 
+def new_host_key():
+    """Generate a host key dict which holds a key value pair used for securing SSH connections."""
+    return {HOST_KEY_LOOKUP: generate_private_key(key_size=4096).decode()}
```

Comment:
I was confused where `generate_private_key` private is coming from until I opened the repo and saw it's coming from the tls_certificates charm library. It generates an RSA key which is worth knowing here. I'd add a comment explaining where this function comes from the and the type of key it's generating.